compiler/src/dmd/location.d

```diff
@@ -108,7 +108,7 @@ nothrow:
      * Params:
      *   name = file name for location, null for no file name
      */
-    extern (C++) void filename(const(char)* name)
+    extern (C++) void filename(const(char)* name) @trusted
```

Comment:
The one `@trusted` addition

Reply:
The filename array is `__gshared`

Reply:
I'm asking because both seem actionable.

Reply:
Is there any reason why it's `_gshared` ?

Reply:
Because it's faster than accessing TLS and dmd is single-threaded anyway

Reply:
> I'm asking because both seem actionable.

Not sure what you mean with 'both' and 'actionable', do you have a suggested change?

Reply:
That former reply was related to my other comment: "What makes it unsafe? The call to push or the cast?"